waspc/data/Generator/templates/server/src/config.js

```diff
@@ -38,5 +37,5 @@ const config = {
   }
 }
 
-const resolvedConfig = _.merge(config.all, config[env])
+const resolvedConfig = {...config.all, ...config[env]}
```

Comment:
Hm, looks like there's a bug here. Object spread only performs a shallow copy/override, while `_merge` is recursive. Try comparing the following outputs (you can use the [REPL in lodash's docs](https://lodash.com/docs/4.17.15#merge)):
```javascript
const config1 = {
  a: [{ b: 2 }, { d: 4 }],
  e: { f: 1, g: 2 }
};
 
const config2 = {
  a: [{ c: 3 }, { e: 5 }],
  e: { g: 3, h: 4}
};
 
console.log(_.merge(config1, config2));
console.log({...config1, ...config2});
```

Reply:
Hey! Good point, I looked at the configs being merged and they seemed shallow so I figured it was enough. I'll circle back and allow recursive merging.

Reply:
Hey, so I'll be going on vacations and won't be around for a while to get to this, so I'm leaving my 2c.

The fix is not trivial and there are a bunch of gotchas, good read here: https://stackoverflow.com/a/28248548

My suggestion is to just use Lodash's es version of just this function. [It's small](https://bundlephobia.com/package/lodash.merge@4.6.2), battle-tested and still gives users the flexibility to choose their own Lodash version separate from this. You could even give it an alias if you want in the package.json.

Good luck with the project! Cheers.

Reply:
Thanks for the input!

Yes, I think you're right and we'll most likely do as you suggested.
When we start supporting TypeScript, we'll leverage its type system to impose additional constraints on config objects (i.e., restrict them to only normal value-based objects). I thought we could have a simpler merge function because we'd be doing this anyway (forgot to mention this constraint, apologies), but I'll feel much more comfortable once the type system can enforce and document this for us.

In the meantime, your solution makes the most sense, I think. Check back when you come back from vacation and we might have an unresolved issue waiting for you (if you remain interested, of course :)).

I'll update this thread with details. Cheers!